### PR TITLE
[WWST-1994] New DTH Z-Wave Basic Window Shade (Aeotec Nano Shutter)

### PR DESCRIPTION
**Pull request by PKacprowiczS**
@greens @dkirker @MMateusiakS @KKlimczukS @DCzarneckaS @MGoralczykS

### CODE REVIEW

File: devicetypes/smartthings/zwave-basic-window-shade.src/zwave-basic-window-shade.groovy [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+ */
+
+metadata {
+	definition (name: "Z-Wave Basic Window Shade", namespace: "smartthings", author: "SmartThings") {
+		capability "Stateless Curtain Power Button"
+		capability "Configuration"
+		capability "Actuator"
+		capability "Health Check"
+
+		command "open"
```
Comment (greens) [inline devicetypes/smartthings/zwave-basic-window-shade.src/zwave-basic-window-shade.groovy:24]: @dkirker The `Stateless Curtain Power Button` capability is only proposed, but these commands are part of it, so they shouldn't need to be defined separately, right?

Comment (greens) [reply devicetypes/smartthings/zwave-basic-window-shade.src/zwave-basic-window-shade.groovy:24]: Based on the commented-out code below @PKacprowiczS I'm guessing it didn't work without them, though?

File: devicetypes/smartthings/zwave-basic-window-shade.src/zwave-basic-window-shade.groovy [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+	result
+}
+
+def ping() {
+	secure(zwave.switchMultilevelV3.switchMultilevelGet())
+}
+
+def installed() {
+	log.debug "Installed ${device.displayName}"
+	sendEvent(name: "checkInterval", value: 2 * 15 * 60 + 2 * 60, displayed: false, data: [protocol: "zwave", hubHardwareId: device.hub.hardwareID, offlinePingable: "1"])
```
Comment (greens) [inline devicetypes/smartthings/zwave-basic-window-shade.src/zwave-basic-window-shade.groovy]: I'd remove `offlinePingable` as this isn't really a crucial device.

**Review by greens — COMMENTED** [review]
I almost wonder if it's not just better to use the general WindowShade capability and just never send the "open" or "closed" states? I know it's at least mapped better in ocf.

File: devicetypes/smartthings/zwave-basic-window-shade.src/zwave-basic-window-shade.groovy (hunk 1 above)
Comment (dkirker) [reply devicetypes/smartthings/zwave-basic-window-shade.src/zwave-basic-window-shade.groovy:24]: So, it appears that `Stateless Curtain Power Button` has the one command `setButton(button)` which takes one of `["open","close","pause"]`.

**Comment by dkirker** [comment]
I believe part of the reason for using the current capability was for the "pause" command. This should work if the `setButton(button)` command is implemented that accepts an enum of `["open","close","pause"]`. I'd leave the `open()`, `close()`, and `pause()` commands in too for Classic and just have `setButton(button)` call them as necessary.

Also, you will want to do a `sendEvent(name: "availableCurtainPowerButtons", value: ["open", "close", "pause"])`

**Review by dkirker — CHANGES_REQUESTED** [review]
Once my comments above have been added, the device should work from OneApp. However, it seems that Stateless Curtain Power Button has not been mapped yet.

File: devicetypes/smartthings/zwave-basic-window-shade.src/zwave-basic-window-shade.groovy (hunk 1 above)
Comment (greens) [reply devicetypes/smartthings/zwave-basic-window-shade.src/zwave-basic-window-shade.groovy:24]: ah, makes sense

**Review by greens — DISMISSED** [review]

**Review by dkirker — DISMISSED** [review]
Looks good. I pinged the proper team on the ticket for the last step for the capability integration.

**Comment by greens** [comment]
Do we know if stateless curtain power button works yet?

File: devicetypes/smartthings/zwave-basic-window-shade.src/zwave-basic-window-shade.groovy [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+}
+
+def ping() {
+	secure(zwave.switchMultilevelV3.switchMultilevelGet())
+}
+
+def installed() {
+	log.debug "Installed ${device.displayName}"
+	sendEvent(name: "checkInterval", value: 2 * 15 * 60 + 2 * 60, displayed: false, data: [protocol: "zwave", hubHardwareId: device.hub.hardwareID])
+	sendEvent(name: "availableCurtainPowerButtons", value: ["open", "close", "pause"])
```
Comment (greens) [inline devicetypes/smartthings/zwave-basic-window-shade.src/zwave-basic-window-shade.groovy]: I _think_ you just need to make this JSON

**Comment by PKacprowiczS** [comment]
@greens After every _pause()_ call device is sending by itself not-so-accurate report about its current state (e.g. when transiting from close -> pause, I've received _closing_ event) so I had to create this flag to ignore next response after calling _pause()_

File: devicetypes/smartthings/zwave-basic-window-shade.src/zwave-basic-window-shade.groovy (hunk 3 above)
Comment (greens) [reply devicetypes/smartthings/zwave-basic-window-shade.src/zwave-basic-window-shade.groovy]: 👀 @PKacprowiczS

Comment (greens) [reply devicetypes/smartthings/zwave-basic-window-shade.src/zwave-basic-window-shade.groovy]: so based on your comment on the ticket, this _doesn't_ need to be JSON?

Comment (PKacprowiczS) [reply devicetypes/smartthings/zwave-basic-window-shade.src/zwave-basic-window-shade.groovy]: No, it doesn't.

File: devicetypes/smartthings/zwave-basic-window-shade.src/zwave-basic-window-shade.groovy [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+	state.direction = direction ? direction : "Normal"
+}
+
+def configure() {
+	log.debug "Configure..."
+	response(secure(zwave.configurationV1.configurationSet(parameterNumber: 80, size: 1, scaledConfigurationValue: 1)))
+}
+
+private secure(cmd) {
+	if(zwaveInfo.zw.endsWith("s")) {
```
Comment (greens) [inline devicetypes/smartthings/zwave-basic-window-shade.src/zwave-basic-window-shade.groovy]: change this to `contains`

File: devicetypes/smartthings/zwave-basic-window-shade.src/zwave-basic-window-shade.groovy [hunk 5]
```diff
@@ -0,0 +1,162 @@
...
+			state "default", label:'Pause', action:"pause"
+		}
+
+		details(["open","close","pause"])
+	}
+
+	preferences {
+		input (
+				title: "Working direction",
+				description: "In case wiring is wrong, this setting can be changed to fix setup without any manual maintenance.",
```
Comment (greens) [inline devicetypes/smartthings/zwave-basic-window-shade.src/zwave-basic-window-shade.groovy:49]: Would it make our lives easier if this was a boolean that asked whether or not you want to reverse working direction?

File: devicetypes/smartthings/zwave-basic-window-shade.src/zwave-basic-window-shade.groovy (hunk 3 above)
Comment (dkirker) [reply devicetypes/smartthings/zwave-basic-window-shade.src/zwave-basic-window-shade.groovy]: The capability is defined as JSON:

```
  availableCurtainPowerButtons:
    schema:
      type: object
      additionalProperties: false
      properties:
        value:
          type: array
          items:
            type: string
            enum:
              - open
              - close
              - pause
    type: JSON_OBJECT
```
Please try it as JSON and assuming that works then change it to that. If it doesn't,  there is a bug somewhere. But it should be a JSON array containing items of type string.

Comment (PKacprowiczS) [reply devicetypes/smartthings/zwave-basic-window-shade.src/zwave-basic-window-shade.groovy]: It works for both JSON and array.

**Review by greens — APPROVED** [review]

**Pull request merged**